transport/http2_server.go

```diff
@@ -722,30 +699,29 @@ func (t *http2Server) WriteHeader(s *Stream, md metadata.MD) error {
 	}
 	md = s.header
 	s.mu.Unlock()
-
-	hBuf := bytes.NewBuffer([]byte{}) // TODO(mmukhi): Try and re-use this memory later.
-	hEnc := hpack.NewEncoder(hBuf)
-	hEnc.WriteField(hpack.HeaderField{Name: ":status", Value: "200"})
-	hEnc.WriteField(hpack.HeaderField{Name: "content-type", Value: "application/grpc"})
+	headerFields := make([]hpack.HeaderField, 0, 2) // aleast :status, content-type will be there if none else.
```

Comment:
I think it makes sense to do the math to avoid the reallocation/copy.

Reply:
Same argument as above.